src/main/webapp/script.js

```diff
@@ -249,3 +257,27 @@ function createRating(rating) {
   
   return ratingDiv;
 }
+
+window.onbeforeunload = function() {
+  localStorage.setItem("listings", JSON.stringify(listingsLocalStorage));
+  localStorage.setItem("location", locationQuery);
+  localStorage.setItem("zipcode", document.getElementById('zipCode').value);
+  localStorage.setItem("product", document.getElementById('product').value);
+}
+
+window.onload = function() {
+  listingsLocalStorage = JSON.parse(localStorage.getItem("listings"));
+  locationQuery = localStorage.getItem("location");
+  bounds = new google.maps.LatLngBounds();
+  let zipcode = localStorage.getItem("zipcode");
+  let product = localStorage.getItem("product");
+
+  if(listingsLocalStorage != null && locationQuery != null && zipcode != null){
+    hideEntryContainer();
+    document.getElementById('zipCode').value = zipcode;
+    addResultCardsAndMapToTheScreen(listingsLocalStorage);
+    if (product !== "") {
```

Comment:
Should locationQuery be mentioned in this block somewhere? It's in the if statement, but not used.

Reply:
Even though it is not being used, I do not want the page to load the most recent list of businesses, without have the locationQuery as well

Reply:
Got it. Sgtm.